airflow/models/__init__.py

```diff
@@ -4189,17 +4243,19 @@ def create_dagrun(self,
             conf=conf,
             state=state
         )
+        (tis, edges) = self.create_tis_and_edges(execution_date)
```

Comment:
Can we use the `create_session` here instead:
```
    def create_dagrun(self,
                      run_id,
                      state,
                      execution_date,
                      start_date=None,
                      external_trigger=False,
                      conf=None):
        """
        Creates a dag run from this dag including the tasks associated with this dag.
        Returns the dag run.
        :param run_id: defines the the run id for this dag run
        :type run_id: str
        :param execution_date: the execution date of this dag run
        :type execution_date: datetime
        :param state: the state of the dag run
        :type state: State
        :param start_date: the date this dag run should be evaluated
        :type start_date: datetime
        :param external_trigger: whether this dag run is externally triggered
        :type external_trigger: bool
        :param session: database session
        :type session: Session
        """
        with create_session():
            run = DagRun(
                dag_id=self.dag_id,
                run_id=run_id,
                execution_date=execution_date,
                start_date=start_date,
                external_trigger=external_trigger,
                conf=conf,
                state=state
            )
            (tis, edges) = self.create_tis_and_edges(execution_date)
            session.add(run)

            for ti in tis:
                session.merge(ti)
            for edge in edges:
                session.merge(edge)

        run.dag = self
        run.refresh_from_db()
        return run
```
Although the `run.dag = self` and subsequently the `refresh_from_db` feels a bit fishy to me.

Reply:
agree, fixed this

Reply:
I needed to revert this. Fixing this is a good idea but not in this PR. I try to do this but did this mean I need to change 20+ files for this.
This is something we should do in a new PR.